### ISSUE TITLE
Exposed port with manage network couldn't be reached outside host on Centos7

### ISSUE THREAD

**Issue by easyrasta** [rancher/rancher#4323]
Rancher Version:
**1.0.0**

Docker Version:
**Docker version 1.10.3, build 20f81dd**

OS:
**Centos7 4.5.0-1.el7.elrepo.x86_64 **

Steps to Reproduce: 
Create a stack, add a service with container (eg, elasticsearch) with managed network and exposed port 9200
curl http://localhost:9200 return
```
{
  "name" : "Captain Savage",
  "cluster_name" : "elasticsearch",
  "version" : {
    "number" : "2.3.1",
    "build_hash" : "bd980929010aef404e7cb0843e61d0665269fc39",
    "build_timestamp" : "2016-04-04T12:25:05Z",
    "build_snapshot" : false,
    "lucene_version" : "5.5.0"
  },
  "tagline" : "You Know, for Search"
}
```
curl from other host return 
```
connect to 192.168.249.111 port 9200 failed: Connection refused
```
my iptables, iptables -L -n -t nat
```
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination
CATTLE_PREROUTING  all  --  0.0.0.0/0            0.0.0.0/0
PREROUTING_direct  all  --  0.0.0.0/0            0.0.0.0/0
PREROUTING_ZONES_SOURCE  all  --  0.0.0.0/0            0.0.0.0/0
PREROUTING_ZONES  all  --  0.0.0.0/0            0.0.0.0/0
DOCKER     all  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
OUTPUT_direct  all  --  0.0.0.0/0            0.0.0.0/0
DOCKER     all  --  0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination
CATTLE_POSTROUTING  all  --  0.0.0.0/0            0.0.0.0/0
MASQUERADE  all  --  172.17.0.0/16        0.0.0.0/0
POSTROUTING_direct  all  --  0.0.0.0/0            0.0.0.0/0
POSTROUTING_ZONES_SOURCE  all  --  0.0.0.0/0            0.0.0.0/0
POSTROUTING_ZONES  all  --  0.0.0.0/0            0.0.0.0/0
MASQUERADE  udp  --  172.17.0.2           172.17.0.2           udp dpt:4500
MASQUERADE  udp  --  172.17.0.2           172.17.0.2           udp dpt:500
MASQUERADE  tcp  --  172.17.0.3           172.17.0.3           tcp dpt:9200

Chain CATTLE_POSTROUTING (1 references)
target     prot opt source               destination
ACCEPT     all  --  10.42.0.0/16         169.254.169.250
MASQUERADE  tcp  --  10.42.0.0/16        !10.42.0.0/16         masq ports: 1024-65535
MASQUERADE  udp  --  10.42.0.0/16        !10.42.0.0/16         masq ports: 1024-65535
MASQUERADE  all  --  10.42.0.0/16        !10.42.0.0/16
SNAT       all  -- !10.42.0.0/16         169.254.169.250      mark match 0x3be07 to:10.42.245.255
SNAT       all  -- !10.42.0.0/16         169.254.169.250      mark match 0x25a1e to:10.42.154.142
MASQUERADE  tcp  --  172.17.0.0/16        0.0.0.0/0            masq ports: 1024-65535
MASQUERADE  udp  --  172.17.0.0/16        0.0.0.0/0            masq ports: 1024-65535

Chain CATTLE_PREROUTING (1 references)
target     prot opt source               destination
DNAT       tcp  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL tcp dpt:9200 to:10.42.245.255:9200
MARK       all  -- !10.42.0.0/16         169.254.169.250      MAC 02:95:C2:2F:D7:6F MARK set 0x3be07
MARK       all  -- !10.42.0.0/16         169.254.169.250      MAC 02:95:C2:50:C5:41 MARK set 0x25a1e

Chain DOCKER (2 references)
target     prot opt source               destination
RETURN     all  --  0.0.0.0/0            0.0.0.0/0
DNAT       udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:4500 to:172.17.0.2:4500
DNAT       udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:500 to:172.17.0.2:500
DNAT       tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:9200 to:172.17.0.3:9200

Chain OUTPUT_direct (1 references)
target     prot opt source               destination

Chain POSTROUTING_ZONES (1 references)
target     prot opt source               destination
POST_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]
POST_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]

Chain POSTROUTING_ZONES_SOURCE (1 references)
target     prot opt source               destination

Chain POSTROUTING_direct (1 references)
target     prot opt source               destination

Chain POST_public (2 references)
target     prot opt source               destination
POST_public_log  all  --  0.0.0.0/0            0.0.0.0/0
POST_public_deny  all  --  0.0.0.0/0            0.0.0.0/0
POST_public_allow  all  --  0.0.0.0/0            0.0.0.0/0

Chain POST_public_allow (1 references)
target     prot opt source               destination

Chain POST_public_deny (1 references)
target     prot opt source               destination

Chain POST_public_log (1 references)
target     prot opt source               destination

Chain PREROUTING_ZONES (1 references)
target     prot opt source               destination
PRE_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]
PRE_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]

Chain PREROUTING_ZONES_SOURCE (1 references)
target     prot opt source               destination

Chain PREROUTING_direct (1 references)
target     prot opt source               destination

Chain PRE_public (2 references)
target     prot opt source               destination
PRE_public_log  all  --  0.0.0.0/0            0.0.0.0/0
PRE_public_deny  all  --  0.0.0.0/0            0.0.0.0/0
PRE_public_allow  all  --  0.0.0.0/0            0.0.0.0/0

Chain PRE_public_allow (1 references)
target     prot opt source               destination

Chain PRE_public_deny (1 references)
target     prot opt source               destination

Chain PRE_public_log (1 references)
target     prot opt source               destination
```

iptables -L -n
```
target     prot opt source               destination
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0
INPUT_direct  all  --  0.0.0.0/0            0.0.0.0/0
INPUT_ZONES_SOURCE  all  --  0.0.0.0/0            0.0.0.0/0
INPUT_ZONES  all  --  0.0.0.0/0            0.0.0.0/0
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0
REJECT     all  --  0.0.0.0/0            0.0.0.0/0            reject-with icmp-host-prohibited

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
DOCKER-ISOLATION  all  --  0.0.0.0/0            0.0.0.0/0
DOCKER     all  --  0.0.0.0/0            0.0.0.0/0
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0
FORWARD_direct  all  --  0.0.0.0/0            0.0.0.0/0
FORWARD_IN_ZONES_SOURCE  all  --  0.0.0.0/0            0.0.0.0/0
FORWARD_IN_ZONES  all  --  0.0.0.0/0            0.0.0.0/0
FORWARD_OUT_ZONES_SOURCE  all  --  0.0.0.0/0            0.0.0.0/0
FORWARD_OUT_ZONES  all  --  0.0.0.0/0            0.0.0.0/0
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0
REJECT     all  --  0.0.0.0/0            0.0.0.0/0            reject-with icmp-host-prohibited

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
OUTPUT_direct  all  --  0.0.0.0/0            0.0.0.0/0

Chain DOCKER (1 references)
target     prot opt source               destination
ACCEPT     udp  --  0.0.0.0/0            172.17.0.2           udp dpt:4500
ACCEPT     udp  --  0.0.0.0/0            172.17.0.2           udp dpt:500
ACCEPT     tcp  --  0.0.0.0/0            172.17.0.3           tcp dpt:9200

Chain DOCKER-ISOLATION (1 references)
target     prot opt source               destination
RETURN     all  --  0.0.0.0/0            0.0.0.0/0

Chain FORWARD_IN_ZONES (1 references)
target     prot opt source               destination
FWDI_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]
FWDI_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]

Chain FORWARD_IN_ZONES_SOURCE (1 references)
target     prot opt source               destination

Chain FORWARD_OUT_ZONES (1 references)
target     prot opt source               destination
FWDO_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]
FWDO_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]

Chain FORWARD_OUT_ZONES_SOURCE (1 references)
target     prot opt source               destination

Chain FORWARD_direct (1 references)
target     prot opt source               destination

Chain FWDI_public (2 references)
target     prot opt source               destination
FWDI_public_log  all  --  0.0.0.0/0            0.0.0.0/0
FWDI_public_deny  all  --  0.0.0.0/0            0.0.0.0/0
FWDI_public_allow  all  --  0.0.0.0/0            0.0.0.0/0

Chain FWDI_public_allow (1 references)
target     prot opt source               destination

Chain FWDI_public_deny (1 references)
target     prot opt source               destination

Chain FWDI_public_log (1 references)
target     prot opt source               destination

Chain FWDO_public (2 references)
target     prot opt source               destination
FWDO_public_log  all  --  0.0.0.0/0            0.0.0.0/0
FWDO_public_deny  all  --  0.0.0.0/0            0.0.0.0/0
FWDO_public_allow  all  --  0.0.0.0/0            0.0.0.0/0

Chain FWDO_public_allow (1 references)
target     prot opt source               destination

Chain FWDO_public_deny (1 references)
target     prot opt source               destination

Chain FWDO_public_log (1 references)
target     prot opt source               destination

Chain INPUT_ZONES (1 references)
target     prot opt source               destination
IN_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]
IN_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto]

Chain INPUT_ZONES_SOURCE (1 references)
target     prot opt source               destination

Chain INPUT_direct (1 references)
target     prot opt source               destination

Chain IN_public (2 references)
target     prot opt source               destination
IN_public_log  all  --  0.0.0.0/0            0.0.0.0/0
IN_public_deny  all  --  0.0.0.0/0            0.0.0.0/0
IN_public_allow  all  --  0.0.0.0/0            0.0.0.0/0

Chain IN_public_allow (1 references)
target     prot opt source               destination
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:22 ctstate NEW
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:2375 ctstate NEW

Chain IN_public_deny (1 references)
target     prot opt source               destination

Chain IN_public_log (1 references)
target     prot opt source               destination

Chain OUTPUT_direct (1 references)
target     prot opt source               destination
```

**Comment by easyrasta** [comment]
same probleme if I try elasticsearch from catalog, kopf is not available. (ES isnt open to public)

**Comment by easyrasta** [comment]
I try on:
**ubuntu-15 4.2.0-35-generic**
Works fine.

Problem seem related to **firewalld on centos7**

Here my working ubuntu iptables:
```
chain PREROUTING (policy ACCEPT)
target     prot opt source               destination
CATTLE_PREROUTING  all  --  0.0.0.0/0            0.0.0.0/0
DOCKER     all  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
DOCKER     all  --  0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination
CATTLE_POSTROUTING  all  --  0.0.0.0/0            0.0.0.0/0
MASQUERADE  all  --  172.17.0.0/16        0.0.0.0/0
MASQUERADE  udp  --  172.17.0.2           172.17.0.2           udp dpt:4500
MASQUERADE  udp  --  172.17.0.2           172.17.0.2           udp dpt:500
MASQUERADE  tcp  --  172.17.0.3           172.17.0.3           tcp dpt:9200

Chain CATTLE_POSTROUTING (1 references)
target     prot opt source               destination
ACCEPT     all  --  10.42.0.0/16         169.254.169.250
MASQUERADE  tcp  --  10.42.0.0/16        !10.42.0.0/16         masq ports: 1024-65535
MASQUERADE  udp  --  10.42.0.0/16        !10.42.0.0/16         masq ports: 1024-65535
MASQUERADE  all  --  10.42.0.0/16        !10.42.0.0/16
SNAT       all  -- !10.42.0.0/16         169.254.169.250      mark match 0x25631 to:10.42.153.137
SNAT       all  -- !10.42.0.0/16         169.254.169.250      mark match 0x1023 to:10.42.4.131
MASQUERADE  tcp  --  172.17.0.0/16        0.0.0.0/0            masq ports: 1024-65535
MASQUERADE  udp  --  172.17.0.0/16        0.0.0.0/0            masq ports: 1024-65535

Chain CATTLE_PREROUTING (1 references)
target     prot opt source               destination
DNAT       tcp  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL tcp dpt:9200 to:10.42.153.137:9200
MARK       all  -- !10.42.0.0/16         169.254.169.250      MAC 02:95:C2:A7:C5:79 MARK set 0x25631
MARK       all  -- !10.42.0.0/16         169.254.169.250      MAC 02:95:C2:46:48:61 MARK set 0x1023

Chain DOCKER (2 references)
target     prot opt source               destination
RETURN     all  --  0.0.0.0/0            0.0.0.0/0
DNAT       udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:4500 to:172.17.0.2:4500
DNAT       udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:500 to:172.17.0.2:500
DNAT       tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:9200 to:172.17.0.3:9200
```

**Comment by easyrasta** [comment]
related to #2145 ?

**Comment by easyrasta** [comment]
and a diff before and after a rejected curl on 9200
```
3c3
< 18388   53M ACCEPT     all  --  *      *       0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
---
> 18420   53M ACCEPT     all  --  *      *       0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
13,16c13,16
<  4435   13M DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<  2297   13M DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0
<  1646   13M ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
<  2138  121K ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0
---
>  4442   13M DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>  2299   13M DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0
>  1647   13M ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
>  2143  121K ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0
20,24c20,24
<     3   180 FORWARD_direct  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<     3   180 FORWARD_IN_ZONES_SOURCE  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<     3   180 FORWARD_IN_ZONES  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<     3   180 FORWARD_OUT_ZONES_SOURCE  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<     3   180 FORWARD_OUT_ZONES  all  --  *      *       0.0.0.0/0            0.0.0.0/0
---
>     4   240 FORWARD_direct  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>     4   240 FORWARD_IN_ZONES_SOURCE  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>     4   240 FORWARD_IN_ZONES  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>     4   240 FORWARD_OUT_ZONES_SOURCE  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>     4   240 FORWARD_OUT_ZONES  all  --  *      *       0.0.0.0/0            0.0.0.0/0
26c26
<     3   180 REJECT     all  --  *      *       0.0.0.0/0            0.0.0.0/0            reject-with icmp-host-prohibited
---
>     4   240 REJECT     all  --  *      *       0.0.0.0/0            0.0.0.0/0            reject-with icmp-host-prohibited
28c28
< Chain OUTPUT (policy ACCEPT 7163 packets, 18M bytes)
---
> Chain OUTPUT (policy ACCEPT 7188 packets, 18M bytes)
30c30
< 20055   43M OUTPUT_direct  all  --  *      *       0.0.0.0/0            0.0.0.0/0
---
> 20080   43M OUTPUT_direct  all  --  *      *       0.0.0.0/0            0.0.0.0/0
40c40
<  4435   13M RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0
---
>  4442   13M RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0
44c44
<     3   180 FWDI_public  all  --  eno16780032 *       0.0.0.0/0            0.0.0.0/0           [goto]
---
>     4   240 FWDI_public  all  --  eno16780032 *       0.0.0.0/0            0.0.0.0/0           [goto]
53c53
<     3   180 FWDO_public  all  --  *      +       0.0.0.0/0            0.0.0.0/0           [goto]
---
>     4   240 FWDO_public  all  --  *      +       0.0.0.0/0            0.0.0.0/0           [goto]
63,65c63,65
<     3   180 FWDI_public_log  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<     3   180 FWDI_public_deny  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<     3   180 FWDI_public_allow  all  --  *      *       0.0.0.0/0            0.0.0.0/0
---
>     4   240 FWDI_public_log  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>     4   240 FWDI_public_deny  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>     4   240 FWDI_public_allow  all  --  *      *       0.0.0.0/0            0.0.0.0/0
78,80c78,80
<     3   180 FWDO_public_log  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<     3   180 FWDO_public_deny  all  --  *      *       0.0.0.0/0            0.0.0.0/0
<     3   180 FWDO_public_allow  all  --  *      *       0.0.0.0/0            0.0.0.0/0
---
>     4   240 FWDO_public_log  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>     4   240 FWDO_public_deny  all  --  *      *       0.0.0.0/0            0.0.0.0/0
>     4   240 FWDO_public_allow  all  --  *      *       0.0.0.0/0            0.0.0.0/0
```

**Comment by easyrasta** [comment]
solution is to open port on host:
```
firewall-cmd --permanent --zone=public --add-port=9200/tcp
firewall-cmd --reload
```

**Comment by virtuman** [comment]
same here on centos7. not publishing ports and expose is somewhat fishy, barely works i would call it (90% of the time it doesn't, but if i stay persistent and continue erasing and re-creating container then eventually it exposes it, but publishing ports doesn't open them at all to public access

**Comment by ibuildthecloud** [comment]
Yes, this is an issue that will be fixed shortly.  If the FORWARD chain does not default ACCEPT then the port forwarding breaks.  We will update the FORWARD chain appropriately to allow users to have a default DENY FORWARD chain.

**Comment by deniseschannon** [comment]
@easyrasta @virtuman Would either of you be able to test the fix in v1.1.0-dev2-rc4 for me? 

I have been unable to get an older setup to test this use case, so it'd be great if you could verify for me.

**Comment by aemneina** [comment]
Verified fix with rancher (1.1.0-dev2-rc4 and 1.1.0-dev2-rc6), CentOS (7.2.1511), docker (1.10.3). Environments tested against both cattle and k8s backends.
